### PR TITLE
Add info on the architecture

### LINKED ISSUE #1: Write architecture documentation
Exactly what the title says, write docs so everyone can understand how the architecture works.

### PR DESCRIPTION
**Pull request by Odrakir**
Adds some info to the Readme to explain how the architecture works.

Closes #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,4 +4,196 @@ This is the repository for the official Toggl WatchOS App. It's built in SwiftUI
...
+    public var projects: [Project] = []
+    public var whatever: Int = 0
+}
+
+```
+
+This is how the reducer should look like:
+
+```swift
+public var projectReducer: Reducer<[Project], ProjectAction, API> = Reducer { state, action, Environment in
```
Comment (sofokli1) [inline README.md]: If the reducer signature is `Reducer<[Project], ProjectAction, API>`,  doesn't this mean that `state` in  `{ state, action, Environment in` is an array of projects? What am I getting wrong?

Comment (Odrakir) [reply README.md]: That's the whole idea, this reducer works with only a subpart of the state. I call it `state` but you could also call it `projects`.

By the way, there's an error, the third param should be `api`

Comment (sofokli1) [reply README.md]: if we change state to be called `projects` thus an array (correct me if I got this wrong), how does line 136 work `state.loadingProjects = true` which will become `projects.loadingProjects = true`. right?

Comment (Odrakir) [reply README.md]: Oh, you are right, I added the `loadingProjects` part after the fact as an example and I didn't realise it doesn't make sense at all.

Comment (Odrakir) [reply README.md]: I'll remove that

**Pull request merged**